### PR TITLE
python3Packages.pym3u8downloader: init at 0.1.5

### PR DESCRIPTION
**Pull request by ShamrockLee**
## Description of changes

This pull request packages `pyhon3Packages.pym3u8downloader` and its dependencies.

`python3Packages.pym3u8downloader` has its pytest test cases run in `passthru.tests.pytest`, which makes the tests runnable without network connections, by patching the URLs for the test cases and network connection checks and starts a simple HTTP server before running `pytest`.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pycolorecho/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    owner = "coldsofttech";
+    repo = "pycolorecho";
+    rev = version;
+    hash = "sha256-h/7Wi0x8iLMZpPYekK6W9LTM+2nYJTaKClNtRTzbmdg=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pycolorecho/default.nix:24]:
```suggestion
```

File: pkgs/development/python-modules/pyloggermanager/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchFromGitHub {
+    owner = "coldsofttech";
+    repo = "pyloggermanager";
+    rev = version;
+    hash = "sha256-1hfcmMLH2d71EV71ExKqjZ7TMcqVd1AQrEwJhmEWOVU=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pyloggermanager/default.nix:25]:
```suggestion
```

File: pkgs/development/python-modules/pym3u8downloader/default.nix [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  src = fetchFromGitHub {
+    owner = "coldsofttech";
+    repo = "pym3u8downloader";
+    rev = version;
+    hash = "sha256-Kcvtl4jP2pSiETTKUmuiBsysxaFfd4K/E2/nXY8Vlw8=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pym3u8downloader/default.nix:27]:
```suggestion
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pycolorecho/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "pycolorecho" ];
+
+  meta = {
+    description = "Simple Python package for colorized terminal output";
+    homepage = "https://github.com/coldsofttech/pycolorecho";
+    changelog = "https://github.com/coldsofttech/pycolorecho/blob/${src.rev}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ ];
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pycolorecho/default.nix:33]: I see you left maintainers list empty, you dont want to maintain them?

File: pkgs/development/python-modules/pyloggermanager/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  build-system = [ setuptools ];
+
+  dependencies = [ pycolorecho ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "pyloggermanager" ];
+
+  meta = {
+    description = "A vital logging framework for Python applications";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pyloggermanager/default.nix:32]:
```suggestion
    description = "vital logging framework for Python applications";
```

File: pkgs/development/python-modules/pym3u8downloader/default.nix [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          previousPythonAttrs.postCheck or ""
+          + ''
+            kill -s TERM "$TEST_SERVER_PID"
+          '';
+      });
+    };
+  };
+
+  meta = {
+    description = "A Python class to download and concatenate video files from M3U8 playlists";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pym3u8downloader/default.nix:76]:
```suggestion
    description = "Python class to download and concatenate video files from M3U8 playlists";
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pycolorecho/default.nix (hunk 4 above)
Comment (ShamrockLee) [reply pkgs/development/python-modules/pycolorecho/default.nix:33]: Oops! Thanks for pointing it out!

**Review by Sigmanificient — APPROVED** [review]

File: pkgs/development/python-modules/pyloggermanager/default.nix [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  build-system = [ setuptools ];
+
+  dependencies = [ pycolorecho ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "pyloggermanager" ];
+
+  meta = {
+    description = "vital logging framework for Python applications";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pyloggermanager/default.nix:31]:
```suggestion
    description = "Simple logging framework for Python applications";
```

File: pkgs/development/python-modules/pym3u8downloader/default.nix [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    requests
+  ];
+
+  pythonImportsCheck = [ "pym3u8downloader" ];
+
+  doCheck = false;
+
+  passthru = {
+    tests = {
+      # TODO(@ShamrockLee): Migrate to checkArgs after #272177 gets merged.
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pym3u8downloader/default.nix:37]: The linked PR was closed.

File: pkgs/development/python-modules/pyloggermanager/default.nix (hunk 7 above)
Comment (ShamrockLee) [reply pkgs/development/python-modules/pyloggermanager/default.nix]: I changed it to
```suggestion
    description = "Logging framework for Python applications";
```

**Comment by Aleksanaa** [comment]
@ofborg build python311Packages.pym3u8downloader python312Packages.pym3u8downloader

**Comment by ShamrockLee** [comment]
Both x86_64-linux and aarch64-linux OfBorg builders build fine, x86_64-darwin complains about timeout (after 3600 seconds), and aarch64-darwin build would often not finish in reasonable time.

**Comment by Aleksanaa** [comment]
```
--------- Report for 'aarch64-darwin' ---------
12 packages built:
python311Packages.pycolorecho python311Packages.pycolorecho.dist python311Packages.pyloggermanager python311Packages.pyloggermanager.dist python311Packages.pym3u8downloader python311Packages.pym3u8downloader.dist python312Packages.pycolorecho python312Packages.pycolorecho.dist python312Packages.pyloggermanager python312Packages.pyloggermanager.dist python312Packages.pym3u8downloader python312Packages.pym3u8downloader.dist
```

**Pull request merged**